CodingTracker.AlvinGraham/CodingTracker.AlvinGraham/Utilities.cs

```diff
@@ -0,0 +1,116 @@
+﻿using Spectre.Console;
+
+namespace CodingTracker;
+
+internal static class Utilities
+{
+
+	internal static void ClearScreen(string message)
+	{
+		Console.Clear();
+		Console.WriteLine(message);
+		Console.WriteLine("--------------------------------------");
+	}
+
+	internal static int GetNumber(string message)
+	{
+		string numberInput = AnsiConsole.Ask<string>(message);
+
+		if (numberInput == "0")
+			throw new ArgumentException("Returning to Main Menu");
+
+		var output = Validation.ValidateInt(numberInput, message);
+
+		return output;
+	}
+
+	internal static int GetGoalNumber(string message)
+	{
+		string numberInput = AnsiConsole.Ask<string>(message);
+
+		var output = Validation.ValidateInt(numberInput, message);
+
+		return output;
+	}
+
+	internal static DateTime[] GetDateInputs()
+	{
+		var startDateInput = AnsiConsole.Ask<string>("Input Start Date with the format: dd-mm-yy hh:mm (24 hour clock), or enter 0 to return to main menu.");
+
+		if (startDateInput == "0")
+			throw new ArgumentException("Returning to Main Menu");
+
+		var startDate = Validation.ValidateStartDate(startDateInput);
+
+		var endDateInput = AnsiConsole.Ask<string>("Input End Date with the format: dd-mm-yy hh:mm (24 hour clock), or enter 0 to return to main menu.");
+
+		if (endDateInput == "0")
+			throw new ArgumentException("Returning to Main Menu");
+
+		var endDate = Validation.ValidateEndDate(startDate, endDateInput);
```

Comment:
What if the user entered the start date incorrectly. You are only asking for entering endDate.
You can ask the user for both start and end date until both are correct or user cancel the operation.

Reply:
I'm not exactly sure what you mean here.  I would point out that this method does indeed ask for and validate both startDate and endDate (sequentially) before returning both values as part of a DateTime array.